lib/rucio/web/rest/flaskapi/v1/common.py

```diff
@@ -105,11 +105,13 @@ def parse_scope_name(scope_name):
     :raises ValueError: when scope_name could not be parsed.
     :returns: a (scope, name) tuple.
     """
-    # why again does that regex start with a slash?
-    scope_name = re.match(get_schema_value('SCOPE_NAME_REGEXP'), '/' + scope_name)
-    if scope_name is None:
-        raise ValueError('cannot parse scope and name')
-    return scope_name.group(1, 2)
+    scope_name_regexps = get_scope_name_regexps()
+    for regexp in scope_name_regexps:
+        # why again does that regex start with a slash?
+        result = re.match(regexp, '/' + scope_name)
+        if result is not None:
+            return result.group(1, 2)
+    raise ValueError('cannot parse scope and name')
```

Comment:
shouldn't the method instead take the vo as a parameter and select the correct `scope_name` regex from that? Also please keep the documentation of the function up-to-date.

Reply:
Yes, that's probably a better approach for the flask version. (It wasn't possible to do it that way for webpy since the URLs have to be registered at start up). I will make the change.

Reply:
not all endpoints have the `request.environ['vo']` set, because that is set with the same hook that checks authentication. You could get the header manually in those cases.

Reply:
I've done that for the redirects.